File: langgraph/data/equiv/116.py
def arrange_by_binary_ones(sequence):
    def count_ones_in_binary(number):
        return bin(number).count('1')
    
    return sorted(sorted(sequence), key=count_ones_in_binary)

# method name: sort_array -> arrange_by_binary_ones
# parameter name: arr -> sequence
# introduced a helper function count_ones_in_binary to count '1's in binary representation
# used the helper function in the key argument of the sorted function

File: langgraph/result/qwen/14b-eq2js/116.js
function arrangeByBinaryOnes(sequence) {
    function countOnesInBinary(number) {
        return number.toString(2).split('0').join('').length;
    }
    
    return sequence.slice().sort((a, b) => countOnesInBinary(a) - countOnesInBinary(b));
}

// method name: sortArray -> arrangeByBinaryOnes
// parameter name: arr -> sequence
// introduced a helper function countOnesInBinary to count '1's in binary representation
// used the helper function in the compare function of the sort method